### PR TITLE
[CAS-1303] Make retryPolicy immutable and set via Builder

### PR DESCRIPTION
**Pull request by amitkma**
https://stream-io.atlassian.net/browse/CAS-1303

### 🎯 Goal

The goal of this PR is to make `retryPolicy` immutable and that should be set only with the Builder.

### 🛠 Implementation details

All mutable vars of `retryPolicy` are converted into `val`. A new function `retryPolicy()` is added to Builders to customize the retry policy while creating ChatDomain instance.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

**Review by ogkuzmin — COMMENTED** [review]
Nice job. One thing. Can you add retryPolicy to `io.getstream.chat.android.offline.experimental.plugin.Config` and set retry policy to the `ChatDomain` in `OfflinePlugin::init`

**Review by ogkuzmin — APPROVED** [review]

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt [hunk 1]
```diff
@@ -91,7 +91,7 @@ public sealed interface ChatDomain {
     public val banned: LiveData<Boolean>
 
     /** The retry policy for retrying failed requests */
-    public var retryPolicy: RetryPolicy
```
Comment (zsmb13) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt:94]: Is there a good reason why we're not just deprecating the setter first, and introducing a breaking change instead?

Comment (ogkuzmin) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt:94]: We don't provide any documentation for that. We don't even use it everywhere for retries inside. It's weird also to reset it after ChatDomain is configured already. I have doubts that it's used by anyone currently.

Comment (zsmb13) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt:94]: Given that setting that property was previously the only way to set this, I'd be happier if we just deprecated it at first as we introduced the new API that replaces it, but if you feel like this'd be a super niche feature, I'm okay with doing the breaking change too, given that it's clearly highlighted in the changelog

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**